homeassistant/components/sensor/miflora.py

```diff
@@ -70,17 +69,22 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         backend = GatttoolBackend
     _LOGGER.debug('Miflora is using %s backend.', backend.__name__)
 
-    cache = config.get(CONF_CACHE)
+    cache = config.get(CONF_SCAN_INTERVAL).total_seconds()
     poller = miflora_poller.MiFloraPoller(
         config.get(CONF_MAC), cache_timeout=cache,
         adapter=config.get(CONF_ADAPTER), backend=backend)
     force_update = config.get(CONF_FORCE_UPDATE)
     median = config.get(CONF_MEDIAN)
-    poller.ble_timeout = config.get(CONF_TIMEOUT)
-    poller.retries = config.get(CONF_RETRIES)
 
     devs = []
 
+    try:
+        with async_timeout.timeout(9):
+            yield from hass.async_add_job(poller.fill_cache)
```

Comment:
Use `await`.

Reply:
Use `hass.async_add_executor_job` instead of `hass.async_add_job`.

Reply:
Fixed.